lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/driver/TaskStatusAlarmHandler.java

```diff
@@ -0,0 +1,99 @@
+import java.util.logging.Logger;
+
+/*
+ * Class that gets that status of the tasks from Azure Batch for the job that is currently in progress
+ * and notifies REEF of the status.
+ */
+final class TaskStatusAlarmHandler implements EventHandler<Alarm>, AutoCloseable {
+
+  private final AzureBatchHelper azureBatchHelper;
+  private final InjectionFuture<AzureBatchResourceManager> azureBatchResourceManager;
+  private final InjectionFuture<REEFEventHandlers> reefEventHandlers;
+
+  private static final Logger LOG = Logger.getLogger(AzureBatchResourceLaunchHandler.class.getName());
+
+  @Override
+  public void onNext(final Alarm alarm) {
+    String jobId = this.azureBatchHelper.getAzureBatchJobId();
+    List<CloudTask> allTasks = this.azureBatchHelper.getTaskStatusForJob(jobId);
+
+    // Reschedule alarm again if there are container requests.
+    if (this.azureBatchResourceManager.get().containerRequestCount() > 0) {
+      LOG.log(Level.FINEST, "Reschedule timer since there are container requests", jobId);
+      this.reefEventHandlers.get().scheduleAlarm();
+    } else {
+      LOG.log(Level.INFO, "Not rescheduling timer since there are no container requests", jobId);
+    }
+
+    // Report status if the task has an associated active container.
+    LOG.log(Level.FINER, "Found {0} tasks from job id {1}", new Object[]{allTasks.size(), jobId});
+    LOG.log(Level.FINEST, "active container list: {0}",
+        this.azureBatchResourceManager.get().activeContainerList());
+    for (CloudTask task : allTasks) {
+      State reefTaskState = TaskStatusMapper.getReefTaskState(task);
+      LOG.log(Level.FINEST, "status for Task Id: {0} is [Azure Batch Status]:{1}, [REEF status]:{2}",
+          new Object[]{task.id(), task.state().toString(), reefTaskState});
+      if (this.azureBatchResourceManager.get().isContainerActive(task.id())) {
+        ResourceStatusEvent resourceStatusEvent = ResourceStatusEventImpl.newBuilder()
+            .setIdentifier(task.id())
+            .setState(reefTaskState)
+            .build();
+        LOG.log(Level.FINEST, "Reporting status for Task Id: {0} is [Azure Batch Status]:{1}, [REEF status]:{2}",
+            new Object[]{task.id(), task.state().toString(), reefTaskState});
+        this.reefEventHandlers.get().onResourceStatus(resourceStatusEvent);
+      } else {
+        LOG.log(Level.FINEST,
+            "No active container in resource manager for Task: {0}. Not sending update.", task.id());
+      }
+    }
+  }
+
+  @Inject
+  private TaskStatusAlarmHandler(
+      final InjectionFuture<REEFEventHandlers> reefEventHandlers,
+      final AzureBatchHelper azureBatchHelper,
+      final InjectionFuture<AzureBatchResourceManager> azureBatchResourceManager) {
+    this.reefEventHandlers = reefEventHandlers;
+    this.azureBatchHelper = azureBatchHelper;
+    this.azureBatchResourceManager = azureBatchResourceManager;
+  }
+
+  @Override
+  public void close() throws Exception {
```

Comment:
I don't think this will ever be called. Plus, both close() are blank implementations, so I think we can just remove this.
Unless, there is a way (and a need) to cancel scheduled invocation in Clock.

Reply:
done.